plugins/inputs/x509_cert/x509_cert.go

```diff
@@ -0,0 +1,160 @@
+// Package x509_cert reports metrics from an SSL certificate.
+package x509_cert
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/url"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	_tls "github.com/influxdata/telegraf/internal/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const sampleConfig = `
+  ## List certificate sources
+  # sources = ["/etc/ssl/certs/ssl-cert-snakeoil.pem", "tcp://example.org:443"]
+
+  ## Timeout for SSL connection
+  # timeout = 5
+
+  ## Optional TLS Config
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+
+  ## Use TLS but skip chain & host verification
+  # insecure_skip_verify = false
+`
+const description = "Reads metrics from a SSL certificate"
+
+// X509Cert holds the configuration of the plugin.
+type X509Cert struct {
+	Sources []string          `toml:"sources"`
+	Timeout internal.Duration `toml:"timeout"`
+	_tls.ClientConfig
+}
+
+// Description returns description of the plugin.
+func (c *X509Cert) Description() string {
+	return description
+}
+
+// SampleConfig returns configuration sample for the plugin.
+func (c *X509Cert) SampleConfig() string {
+	return sampleConfig
+}
+
+func (c *X509Cert) getCert(location string, timeout time.Duration) ([]*x509.Certificate, error) {
+	u, err := url.Parse(location)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse cert location - %s\n", err.Error())
+	}
+
+	switch u.Scheme {
+	case "https":
+		u.Scheme = "tcp"
+		fallthrough
+	case "udp":
+		fallthrough
+	case "tcp":
+		tlsCfg, err := c.ClientConfig.TLSConfig()
+		if err != nil {
+			return nil, err
+		}
+
+		ipConn, err := net.DialTimeout(u.Scheme, u.Host, timeout)
+		if err != nil {
+			return nil, err
+		}
+		defer ipConn.Close()
+
+		conn := tls.Client(ipConn, tlsCfg)
+		defer conn.Close()
+
+		hsErr := conn.Handshake()
+		if hsErr != nil {
+			return nil, hsErr
+		}
+
+		certs := conn.ConnectionState().PeerCertificates
+
+		return certs, nil
+	case "file":
+		fallthrough
+	default:
```

Comment:
I think default should be an error, but "" could be file, this would avoid schemes like "foo://".

Reply:
Fixed.